tests/python_tests/common.py

```diff
@@ -325,19 +325,39 @@ def get_default_properties():
     }
 
 
+class StreamerWithResults:
+    # Return a streamer which accumulates results in order to compare with results returned from generate.
+    results: List[str] = []
+    def __init__(self):
+        self.results = []
+
+    def accumulate(self, subword) -> bool:
+        self.results.append(subword)
+        return False
+    
+    def get_results(self) -> List[GenerationResult]:
+        streaming_result = GenerationResult()
+        streaming_result.m_generation_ids = [''.join(self.results)]
+        return [streaming_result]
+    
+    def reset(self):
+        self.results = []
+
+
+
 def run_llm_pipeline(
     models_path : Path,
     prompts: List[str],
     generation_config : GenerationConfig,
-    use_cb : bool = False
+    use_cb : bool = False,
+    streamer: Callable | StreamerBase = None
```

Comment:
why streamer is None by default? 
The idea is to enable comparison streamer output vs generate output for all tests which use `run_llm_pipeline`

Reply:
I was afraid that it will fail in case if there is a beam search, multibatch prompt. AFAIK these are the only 2 states when we don't allow streaming. But i'm not sure maybe there are some more new cases introduced.

Enabled streaming results check for every test which is not multibatch and not beam_search. Fingers crossed, let's hope. Nothing will fail.

But `run_llm_pipeline` is not used in tests, it requires many steps to call it, load HF model, convert and save them. When i used it in `test_llm_pipeline` it looked ugly and with many duplicates with `common.py`.  `run_llm_pipeline_with_ref` is better place to add streamer for each test case.